lib/services/role.go

```diff
@@ -3182,7 +3182,11 @@ func (p boolParser) Parse(string) (any, error) {
 // namespace to the specified resource and verb.
 // silent controls whether the access violations are logged.
 func (set RoleSet) CheckAccessToRule(ctx RuleContext, namespace string, resource string, verb string) error {
-	whereParser, err := NewWhereParser(ctx)
+	whereParser, err := NewWhereParser(
+		ctx,
+		// register can_view function if the resource is a session.
+		ConditionalOption(resource == types.KindSession, WithCanViewFunction()),
```

Comment:
Looks like `ConditionalOption` is only used in `lib/services`. Should we unexport it?